### PR TITLE
Bump Rust versions

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
Makes https://github.com/neondatabase/docker-images/pull/6 obsolete

Uses more up-to-date Rust versions to eliminate compile issues that slowly start to appear every now and then.
See https://neondb.slack.com/archives/C039YKBRZB4/p1649953776799949 for more context.

Also unites the Dockerfiles, following the discussion on https://github.com/neondatabase/docker-images/pull/6#issuecomment-1093446123
I have not went with the templating approach yet for simplicity.

One last question remains: do we want to rename `zimg` into `nimg` or something else?

### CODE REVIEW

File: rust/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,92 @@
+ARG RUST_VERSION=1.58
```
Comment (funbringer) [inline rust/Dockerfile]: Wouldn't it be better not to have a default value?

Comment (SomeoneToIgnore) [reply rust/Dockerfile]: On a second thought, it would, removed.

File: README.md [hunk 2]
```diff
@@ -1,11 +1,9 @@
 # docker-images
 
-Docker images that helps build and test Zenith product
+Docker images tp help build and test [Neon](https://github.com/neondatabase/neon) product
```
Comment (funbringer) [inline README.md]: Looks like a typo

**Review by funbringer — APPROVED** [review]
Many thanks!

**Pull request merged**